smt/problems/tests/test_problem_examples.py

```diff
@@ -80,6 +80,30 @@ def test_hier_neural_network(self):
         plt.ylabel("y")
         plt.show()
 
+    @unittest.skipIf(NO_MATPLOTLIB, "Matplotlib not installed")
+    def test_hier_goldstein(self):

```

Comment:
This code is intended to be in the documentation. It is just run to check if there is no runtime error. But currently it is not used. Besides there is no test (I mean no test with assertions) for this new parameter you introduce while you change the SMT API.